bridge/src/observers/nine-chronicles.ts

```diff
@@ -371,25 +368,41 @@ export class NCGTransferredEventObserver
                     );
                 }
 
-                const exchangeFeeRatio =
-                    this._exchangeFeeRatioPolicy.getFee(sender);
-                if (exchangeFeeRatio === false) {
-                    throw new Error(
-                        `Failed to get exchange fee ratio for ${sender}`
-                    );
-                }
+                const exchangeFeeRatio = this._exchangeFeeRatioPolicy.getFee();
 
                 /**
                  * If exchangeFeeRatio == 0.01 (1%), it exchanges only 0.99 (= 1 - 0.01 = 99%) of amount.
                  * Applied Base Fee Policy, base Fee = 10 when Transfer( NCG -> WNCG ) under 1000 NCG
                  */
-                let fee = limitedAmount.greaterThanOrEqualTo(
-                    new Decimal(this._baseFeePolicy.criterion)
-                )
-                    ? new Decimal(
-                          limitedAmount.mul(exchangeFeeRatio).toFixed(2)
-                      )
-                    : new Decimal(this._baseFeePolicy.fee);
+                let fee;
+                if (
+                    !limitedAmount.greaterThanOrEqualTo(
+                        new Decimal(this._baseFeePolicy.criterion)
+                    )
+                ) {
+                    fee = new Decimal(this._baseFeePolicy.fee);
+                } else {
+                    /**
+                     * Set Fee Per Amount Range
+                     */
+                    if (
+                        !limitedAmount.greaterThan(
+                            exchangeFeeRatio.feeRange1.end
+                        )
+                    ) {
```

Comment:
The values of feeRange1.start and feeRange2.{start,end} don't seem to make sense. There doesn't seem to be a clear specification and implementation of how this works when there is a gap between the BASE_FEE_CRITERION and feeRange1.start, or between feeRange1.end and feeRange2.start. It would be better to have multiple fee ratios and only specify the unit of separation between them. Otherwise, I think it would be important to validate the values well.

<!-- feeRange1.start와 feeRange2.{start,end} 의 값이 의미가 없는것 같습니다. BASE_FEE_CRITERION과 feeRange1.start 사이, feeRange1.end와 feeRange2.start 사이에 간격이 있을때 이는 어떻게 동작하는지에 대한 확실한 스펙 및 구현이 없는 것 같습니다. 차라리 여러 fee ratio 들이 있고 이를 구분하는 구분 단위만 명시하게 하는 것도 좋은 것 같습니다. 아니면 값들을 잘 검증해주는 것이 중요할 것 같습니다. -->

Reply:
Updated!
I removed feeRange1.start, feeRange1.end, feeRange2.start, feeRange2.end
and fixed to check range using BASE_FEE_CRITERION, MAXIMUM_NCG.

I added FEE_RANGE_DIVIDER_AMOUNT to determine the amount range
So each fee range gonna be
```
feeRange1: BASE_FEE_CRITERION <= feeRange1 <= FEE_RANGE_DIVIDER_AMOUNT
feeRange2: FEE_RANGE_DIVIDER_AMOUNT < feeRange2 <= MAXIMUM_NCG